src/qml/QFieldLocalDataPickerScreen.qml

```diff
@@ -474,9 +491,12 @@ Page {
 
         text: qsTr("Push to QFieldCloud")
         onTriggered: {
-          QFieldCloudUtils.addPendingAttachments(projectInfo.cloudUserInformation.username, cloudProjectsModel.currentProjectId, [itemMenu.itemPath], cloudConnection, true);
-          platformUtilities.uploadPendingAttachments(cloudConnection);
-          displayToast(qsTr("‘%1’ is being uploaded to QFieldCloud").arg(FileUtils.fileName(itemMenu.itemPath)));
+          QFieldCloudUtils.addPendingAttachments(cloudConnection.userInformation.username, cloudProjectsModel.currentProjectId, [itemMenu.itemPath], cloudConnection, true);
+          cloudConnection.onAllAttachmentsWritten.connect(function handler() {
+              platformUtilities.uploadPendingAttachments(cloudConnection);
+              displayToast(qsTr("‘%1’ is being uploaded to QFieldCloud").arg(FileUtils.fileName(itemMenu.itemPath)));
+              cloudConnection.onAllAttachmentsWritten.disconnect(handler);
+            });
```

Comment:
I think I'd rather have a Connections { target: cloudConnection;  enabled: false; function onAllAttachment... } QML item instead of this javascript driven option. Thoughts?